### DISCUSSION TITLE
A new "none" Hitdef attr would be handy for collision-based activation.

### DISCUSSION THREAD

**Discussion by VinVulpis** [ikemen-engine/Ikemen-GO#1671] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

So, I've taken to using "empty" hitdefs with [ground.type/air.type= none] as detections to activate behaviors.
Something kinda like this: 
![image](https://github.com/ikemen-engine/Ikemen-GO/assets/66293146/e02206df-ce83-481c-8a0e-268e3ea7837e)

I use this so that when this hifdef makes contact, it simply makes the character activate another state with the "p1stateno." I've grown to prefer doing this over using "p2bodydist" for consistency reasons. I'd rather it just be "can I hit them? activate!" I also would like this for ideas I have for expanding my AI in the future, to aid in decision making based on if a helper is in contact with a possible target before the AI character tries something.

I make them unblockable and something like "special throw" so most reversaldefs don't trigger it, but it comes with it's own lil side effects. AI might react oddly to it, post wake up throw invincibility messes with it, if I make it an attack or projectile, this fake hit can be parried. 

### Describe the solution you'd like

The basic solution I'm asking for is a new hitdef attr specifically for fake hits like this. Something like a "none" or "N, NN" or something to that effect. 

That, OR, some other way to detect is collision boxes are touching without the need for a hitdef at all, like a new trigger for it to check if a red box is touching another red or blue box without a hitdef.

### Additional context

I think this would have a lot of utility for things to use as activation triggers or "Area of Effect" type things, also AI utility. Extra points for the AI aspect if there is a way to get data from the possible target of "will this be hit-overridden, will this be dodged by I-frames?" to stop AI from attacking something it can't hit.

If something like it already does exist and I'm not aware of it, please, do tell. 

Thanks much!

### Search terms

hitdef attr

**Comment by potsmugen** [comment, marked as answer]
Type None already existed since Mugen though?

But the nightly Clsn triggers should allow even more flexibility than that.

**Reply by VinVulpis** [reply]
I'll take a look at the new collision triggers to see what I can do. I'm just basically looking for something that checks if two entities are touching without actually "hitting."

**Reply by VinVulpis** [reply]
Yeah it looks like there are new collision detection methods. Perfect!

**Comment by Kiwi1444** [comment]
Can I drain opponent power with such non-interrupting attacks?

**Reply by potsmugen** [reply]
Yes. Sample code:
```
for i = 0; numEnemy - 1; 1 {
	if clsnOverlap(clsn2, enemy($i), ID, clsn2) {
		powerAdd{value: -10; redirectID: enemy($i), ID}
	}
}
```
